openedx/features/redhouse_panel/api/v0/views.py

```diff
@@ -110,34 +113,58 @@ def get(self, request, pk):
                 "address": "Silicon Valley"
             }
         """
-        site = get_object_or_404(Site, pk=pk)
-        site_configuration = SiteConfiguration.objects.filter(site=site).first()
+        site_configuration = configuration_helpers.get_current_site_configuration()
         data = {
-            'name': site.name,
+            'name': site_configuration.get_value('school_name', settings.PLATFORM_NAME),
             'address': site_configuration.values.get('address') if site_configuration else '',
         }
-        return Response(SiteSerializer(data).data)
+        return JsonResponse(SiteSerializer(data).data)
 
 
-class UpdateUserActiveStatus(UpdateAPIView):
+class UserAccountView(viewsets.ModelViewSet):
     """
-    View to change user's active status
-    Raises:
-        NotFound: Raised if user with `pk` provided in `URL` does not exist.
+    Return the list of `Users` associated with the requested site
+
     Example:
-        `PUT/PATCH: /admin-panel/api/v0/users/1/update`
-        body: {"is_active": false}
-        :returns
+        [
             {
-                "id": "1",
-                "is_active": false
-            }
+                id: 2,
+                is_superuser: true,
+                username: "edx",
+                email: "edx@example.com",
+                is_staff: true,
+                is_active: true,
+                profile: {
+                    name: "First Last",
+                    year_of_birth: null
+                },
+                groups: [
+                    {
+                        name: "Redhouse Admin Panel Access"
+                    }
+                ]
+            },
+            ....
+        ]
     """
-    queryset = User.objects.all()
-    serializer_class = UserSerializer
-    authentication_classes = (SessionAuthentication, JwtAuthentication, OAuth2Authentication,)
-    permission_classes = (IsAuthenticated, IsAdminUser)
-
-    def put(self, request, *args, **kwargs):
-        request.data.pop('id', None)
-        return self.partial_update(request, *args, **kwargs)
+
+    authentication_classes = (JwtAuthentication, OAuth2Authentication, SessionAuthentication,)
+    permission_classes = (IsAuthenticated, CanAccessRedhousePanel,)
+    http_method_names = ['get', 'post', 'patch']
+    filter_backends = (filters.SearchFilter,)
+    search_fields = ['profile__name', 'username', 'email']
+    serializer_class = UserAccountSerializer
+    lookup_field = 'username'
+
+    def get_queryset(self):
+        """
+        Return list of Users.
+        """
+
+        # Get the list of the site organizations
```

Comment:
I think these type of functions are self-explanatory and do not require comments/doc-strings, what are your thoughts?

Reply:
Done